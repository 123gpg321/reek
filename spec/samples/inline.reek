[Feature Envy] C#add_type_converter uses type more than self
[Long Method] C#build has approx 61 statements
[Nested Iterators] C#build has nested iterators
[Uncommunicative Name] C#build uses the local variable name 't'
[Feature Envy] C#c uses options more than self
[Uncommunicative Name] C#c uses the method name 'c'
[Feature Envy] C#c2ruby uses type more than self
[Feature Envy] C#c_raw_singleton uses options more than self
[Feature Envy] C#c_singleton uses options more than self
[Long Method] C#generate has approx 33 statements
[Feature Envy] C#generate uses result more than self
[Long Method] C#initialize has approx 17 statements
[Long Method] C#load_cache has approx 7 statements
[Long Method] C#module_name has approx 6 statements
[Nested Iterators] C#module_name has nested iterators
[Long Method] C#parse_signature has approx 15 statements
[Feature Envy] C#ruby2c uses type more than self
[Utility Function] C#strip_comments doesn't depend on instance state
[Large Class] Dir has 31 methods
[Large Class] Module has 34 methods
[Long Method] Module#inline has approx 12 statements
[Feature Envy] Module#inline uses options more than self
