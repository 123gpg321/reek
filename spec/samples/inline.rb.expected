File#self.write_with_backup has approx 6 statements (Long Method)
Inline::C has at least 46 methods (Large Class)
Inline::C#build calls $?.==(0) multiple times (Duplication)
Inline::C#build calls Inline.directory multiple times (Duplication)
Inline::C#build calls io.puts multiple times (Duplication)
Inline::C#build calls io.puts(#endif) multiple times (Duplication)
Inline::C#build calls io.puts(#ifdef __cplusplus) multiple times (Duplication)
Inline::C#build has approx 61 statements (Long Method)
Inline::C#build has the variable name 't' (Uncommunicative Name)
Inline::C#build/block/block has the variable name 'n' (Uncommunicative Name)
Inline::C#build/block/block is nested (Nested Iterators)
Inline::C#c has the name 'c' (Uncommunicative Name)
Inline::C#crap_for_windoze calls Config::CONFIG[libdir] multiple times (Duplication)
Inline::C#generate calls result.sub!((?-mix:\A\n), ) multiple times (Duplication)
Inline::C#generate calls signature[args] multiple times (Duplication)
Inline::C#generate calls signature[args].map multiple times (Duplication)
Inline::C#generate has approx 33 statements (Long Method)
Inline::C#initialize calls stack.empty? multiple times (Duplication)
Inline::C#load_cache has approx 7 statements (Long Method)
Inline::C#module_name has approx 6 statements (Long Method)
Inline::C#module_name refers to md5 more than self (Feature Envy)
Inline::C#module_name/block has the variable name 'm' (Uncommunicative Name)
Inline::C#module_name/block has the variable name 'x' (Uncommunicative Name)
Inline::C#parse_signature has approx 15 statements (Long Method)
Inline::C#parse_signature is controlled by argument raw (Control Couple)
Inline::C#parse_signature/block has the variable name 'x' (Uncommunicative Name)
Inline::C#strip_comments doesn't depend on instance state (Utility Function)
Inline::C#strip_comments refers to src more than self (Feature Envy)
Inline::self.rootdir calls env.nil? multiple times (Duplication)
Inline::self.rootdir has approx 9 statements (Long Method)
Module#inline calls Inline.const_get(lang) multiple times (Duplication)
Module#inline has approx 12 statements (Long Method)
Module#inline is controlled by argument options (Control Couple)
